igcollect/rsyslog_impstats.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+"""igcollect - Rsyslog impstats parser
+
+Copyright (c) 2019 InnoGames GmbH
+"""
+
+import re
+from argparse import ArgumentParser
+from time import time
+
+def parse_log():
+    actions = {}
+    args = parse_args()
+
+    with open(args.filename, 'r') as pstats_file:
+        for line in pstats_file:
+            lastmatch = re.findall(r'(action-[0-9]+)(.*discarded\.full.*)', line)
+            if lastmatch:
+                actions [lastmatch[0][0]] = lastmatch[0][1]
+
+    # Extract the stats we need for the last matched line of each action
+    result = []
+
+    for action in actions: 
+        stats_action = action.replace("-","")
+        stats_base = actions[action].split(': ')[1].split()
+        stats_keyvalue_size = stats_base[1].split('=')
+        stats_keyvalue_enqueued = stats_base[2].split('=')
+        stats_keyvalue_discarded = stats_base[4].split('=')
+
+        result.append({
+            "name": stats_action,
+            stats_keyvalue_size[0]: stats_keyvalue_size[1],
+            stats_keyvalue_enqueued[0]: stats_keyvalue_enqueued[1],
+            stats_keyvalue_discarded[0].replace(".", "_"): stats_keyvalue_discarded[1],
+        })
+
+    return result
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='rsyslog')
+    parser.add_argument('--filename', default='/var/log/pstats')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    stats = parse_log()
+
+    # Define desired fields for output
+    used_fields = [
```

Comment:
Put it to arguments?

Reply:
Added in commit https://github.com/innogames/igcollect/pull/128/commits/34c3003426547aaaaf84e57ee5ff2049eec46e9d